src/edgepi/peripherals/gpio.py

```diff
@@ -23,10 +24,12 @@ def open_gpio(self, pin_num: int = None, pin_dir: str = None, pin_bias: str = No
             pin_bias (str): bias direction
         """
         try:
+            GpioDevice.lock_gpio.acquire()
             self.gpio = GPIO(self.gpio_fd, pin_num, pin_dir, bias=pin_bias)
             yield self.gpio
         finally:
             self.gpio.close()
+            GpioDevice.lock_gpio.release()
```

Comment:
you must put gipi.close in try-finally and in finally release lock, in case close raises exception, otherwise lock may not be released.

Reply:
something like this:
```
 try:
      GpioDevice.lock_gpio.acquire()
      self.gpio = GPIO(self.gpio_fd, pin_num, pin_dir, bias=pin_bias)
      yield self.gpio
finally:
    try:
        self.gpio.close()
    except:
        raise
    finally:
         GpioDevice.lock_gpio.release()
```
note that you need that except to raise the exception again otherwise your exception will not bubble up.

Reply:
[247dcfe](https://github.com/EdgePi-Cloud/edgepi-python-sdk/pull/399/commits/247dcfe655e6fd6f385a4138bb51a21f0aa2799e)